src/client.rs

```diff
@@ -619,21 +626,21 @@ impl Web3 {
         let start = Instant::now();
         loop {
             delay_for(Duration::from_secs(1)).await;
-            match self.eth_get_transaction_by_hash(tx_hash.clone()).await {
+            match self.eth_get_transaction_by_hash(tx_hash).await {
                 Ok(maybe_transaction) => {
                     if let Some(transaction) = maybe_transaction {
                         // if no wait time is specified and the tx is in a block return right away
-                        if blocks_to_wait.clone().is_none() && transaction.block_number.is_some() {
+                        if blocks_to_wait.is_none() && transaction.block_number.is_some() {
                             return Ok(transaction);
                         }
                         // One the tx is in a block we start waiting here
                         else if let (Some(blocks_to_wait), Some(tx_block)) =
-                            (blocks_to_wait.clone(), transaction.block_number.clone())
+                            (blocks_to_wait, transaction.block_number)
                         {
                             let current_block = self.eth_block_number().await?;
                             // we check for underflow, which is possible on testnets
-                            if current_block > blocks_to_wait
-                                && current_block - blocks_to_wait >= tx_block
+                            if (current_block > blocks_to_wait)
```

Comment:
why the braces after IF keyword?

Reply:
I personally always use explicit parenthesis. Sometimes I forget what order of operations Rust uses, or read an especially long if statement like this one the wrong way, so I think it is better to use explicit parenthesis.

Reply:
if you insist